pydatastructs/trees/binary_trees.py

```diff
@@ -415,6 +415,126 @@ def rank(self, x):
             walk = p
         return r
 
+    def _simple_path(self, key, root, path):
+        """
+        Utility funtion to find the simple path between root and node.
+        """
+        if root is None:
+            return False
+        path.append(root)
+        if self.tree[root].key == key:
+            return True
+
+        if self._simple_path(key, self.tree[root].left, path) or \
+            self._simple_path(key, self.tree[root].right, path):
+            return True
+
+        path.pop()
+        return False
+
+    def simple_path(self, key):
+        """
+        Return the intermediate nodes between root and node with given key
+
+        Parameters
+        ==========
+
+        key: Node.key
+            key of the node for which path to be found
+
+        Returns
+        =======
+
+        list
+            List of intermediate nodes.
+            Empty list if node is not present.
+
+        """
+        path = []
+        self._simple_path(key, self.root_idx, path)
+        return path
+
+    def lowest_common_ancestor(self, j, k, algorithm=1):
+
+        """
+        Computes the lowest common ancestor of two nodes.
+
+        Parameters
+        ==========
+
+        j: Node.key
+            key of first node
+        k: Node.key
+            key of second node
+        algorithm: int
+            The algorithm to be used for computing the
+            lowest common ancestor.
+            Optional, by default uses algorithm 1.
+
+            1 -> Determines the lowest common ancestor by finding
+                    the first intersection of the paths from v and w
+                    to the root.
+
+            2 -> Modifed version of the algorithm given in the
+                    following publication,
+                    D. Harel. A linear time algorithm for the
+                    lowest common ancestors problem. In 21s
+                    Annual Symposium On Foundations of
+                    Computer Science, pages 308-319, 1980.
+
+        Returns
+        =======
+
+        int
+            The index of the lowest common ancestor in the tree.
```

Comment:
IMO, returning `key` is better in case of LCA.

Reply:
Okay. Ill do that.